astroquery/esa/jwst/tests/test_jwsttap.py

```diff
@@ -0,0 +1,1088 @@
+
+    def test_query_region(self):
+        connHandler = DummyConnHandler()
+        tapplus = TapPlus("http://test:1111/tap", connhandler=connHandler)
+        tap = JwstClass(tapplus)
+        # Launch response: we use default response because the
+        # query contains decimals
+        responseLaunchJob = DummyResponse()
+        responseLaunchJob.set_status_code(200)
+        responseLaunchJob.set_message("OK")
+        jobDataFile = data_path('job_1.vot')
+        jobData = utils.read_file_content(jobDataFile)
+        responseLaunchJob.set_data(method='POST',
+                                   context=None,
+                                   body=jobData,
+                                   headers=None)
+        # The query contains decimals: force default response
+        connHandler.set_default_response(responseLaunchJob)
+        sc = SkyCoord(ra=29.0, dec=15.0, unit=(u.degree, u.degree),
+                      frame='icrs')
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc)
+            print(err.value.args[0])
+        assert "Missing required argument: 'width'" in err.value.args[0]
+
+        width = Quantity(12, u.deg)
+        height = Quantity(10, u.deg)
+
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width)
+        assert "Missing required argument: 'height'" in err.value.args[0]
+
+        # Test observation_id argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, observation_id=1)
+        assert "observation_id must be string" in err.value.args[0]
+
+        # Test cal_level argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, cal_level='a')
+        assert "cal_level must be either 'Top' or an "\
+               "integer" in err.value.args[0]
+
+        # Test only_public
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, only_public='a')
+        assert "only_public must be boolean" in err.value.args[0]
+
+        # Test dataproduct_type argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, prod_type=1)
+        assert "prod_type must be string" in err.value.args[0]
+
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, prod_type='a')
+        assert "prod_type must be one of: " in err.value.args[0]
+
+        # Test instrument_name argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, instrument_name=1)
+        assert "instrument_name must be string" in err.value.args[0]
+
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height,
+                             instrument_name='a')
```

Comment:
again, please have a test with a valid input, too. And I might have missed a few kwargs where you test for invalid inputs only, it would be great to have a test for the valid ones, too. It's OK, to leave it for the remote tests though rather than mock it.

Reply:
Tested with valid inputs after all assertions